connectors/kafka-connector/src/main/scala/zio/connect/kafka/KafkaConnector.scala

```diff
@@ -0,0 +1,30 @@
+package zio.connect.kafka
+
+import org.apache.kafka.clients.producer.{ProducerRecord}
+import org.apache.kafka.common.Uuid
+import zio.Trace
+import zio.connect.kafka.KafkaConnector._
+import zio.kafka.consumer._
+import zio.stream._
+
+trait KafkaConnector {
+  def createTopic(implicit trace: Trace): ZSink[Any, Throwable, NewTopic, Nothing, Unit]
+
+  def listTopics(implicit trace: Trace): ZStream[Any, Throwable, TopicListing]
+
+  def read(topic: => String)(implicit trace: Trace): ZStream[Any, Throwable, CommittableRecord[String, String]]

```

Comment:
I think it's better not use CommitableRecord type here. It ties the api to zio-kafka. A custom type that in the live implementation is converted to a CommitableRecord would be preferable.

Reply:
I'm not quite sure what's the purpose of these connectors. Make it more simpler and/or user friendly to use different data sources? Because if that then maybe we should make a type parameter here and require some converters to convert CommittableRecord to the final format. Now zio-kafka already uses this new object and if we add another conversion and the user in the end will use another one for the final result it may add a bit of an overhead. I don't know if that matters but might worth to think about this.